main.tf

```diff
@@ -36,6 +36,9 @@ resource "ibm_is_vpc_address_prefix" "address_prefixes" {
   cidr     = each.value.cidr
 }
 
+data "ibm_is_vpc_address_prefixes" "get_address_prefixes" {
```

Comment:
I suspect that this data source is going to return immediately after the vpc is created, but before any of the ibm_is_vpc_address_prefix resource block are executed. In such case, the result in data source will be missed the prefixes address through the resource. Either add a dependency here, or lookup in the ibm_is_vpc_address_prefix resource if it exists (in the logic at address in data.ibm_is_vpc_address_prefixes.get_address_prefixes.address_prefixes )